### ISSUE TITLE
Time management

### ISSUE THREAD

**Issue by RobStallion** [RobStallion/chess-elm#6]
Test time management apps so I can learn to manage my time more effectively. 

#### I would first like to be able to **EASILY** identify how much time I spend doing 'anything' in a given amount of time
How much time I spend cooking in a day, how much time I spend replying to issues in a working week, how much time I spend watching TV in a month, how much time I work on personal projects, etc...

Easily is **KEY!!!!!!!!!!!!!!!!!!!!!!!!!!!!!** to this. I **NEED** to be able to do this with minimal effort. So little effort that I don't even think about it otherwise I won't want to do it.

#### My hope is that this will allow me to easily 'rank' the things I have done in an amount of time (day, week, month). 

The reason I want to rank my tasks is so that I can focus on things that are more important. 

There are things that are important, for example seeing friends, but I do not want to do that everyday for 5 hours a day. I also think personal 'unwinding' time is important (watching tv, playing ps etc...) but are these things taking up too much time and stopping me from doing other things I want to achieve, like **BUILDING THIS APP**?

### Toggl

Going to use this app first and see what I think of it. Will capture my thoughts in this issue.

**Comment by RobStallion** [comment]
I don't think I like toggl 🙄 

It's got some pretty annoying features (I'll list them a bit later)

But something it has already shown me is that I want to improve my speed when it comes to writing examples and readmes. I have been working on https://github.com/dwyl/learn-elixir/issues/105 and barely any of the code I wrote was something new to me.

There was actually only one line of code that I had to work out which was one in the travis yml...
```yml
before_install:
  - cd codecov_example
```
(technically 2 lines but who cares)

I estimated that this would take an hour (I thought it could take longer but I thought at most 2. I wanted an ambitious (for myself) goal).

![image](https://user-images.githubusercontent.com/15571853/52061981-86817100-2567-11e9-81e6-41543c3109f4.png)

It turns out it took about 3 hours and that is before any code review has been done on it. I am not sure yet if this is a problem with me estimating badly, me being too slow when it comes to writing, a combo of both or something else. 

No matter what it's good that it has been highlighted

**Comment by RobStallion** [comment]
Just added another 18 minutes to this 1HOUR task 🙄

**Comment by RobStallion** [comment]
This also seems to have pointed out that I am a bit of a slow reader as I feel like I am spending quite a lot of time reading issues. If there were LOADS of issues then the time would be justified but I think I am just a touch on the slow side.

Not sure how to improve this though 😕 

I may end up making this issue it's own repo because it looks like it may turn into a place where I track a lot of thoughts

**Comment by nelsonic** [comment]
@RobStallion _trust me_, I read "slower" than you. ⏳ 
But act of _tracking_ the time a task/activity is taking is not the "enemy" (_even if the UX is "meh!"_)
_Knowing_ that something is taking longer than you _expected/estimated_ is a _really_ good insight.

![image](https://user-images.githubusercontent.com/194400/52068133-f7c72100-2573-11e9-9911-d552ad858cde.png)

It gives you a chance to improve your _focus_ on the task, define more realistic estimates and/or "break down" the task into smaller chunks (_e.g with a checklist or sub-issues_)

**Comment by RobStallion** [comment]
@nelsonic I am loving the fact that after 2.5 days of using this tool, it already seems to be helping me highlight things I can work and improve on.

I just need to figure out how to go about doing that. 

I also need to get a better understanding of what skills I can improve (so I finish tasks in the time I estimated) and what is just bad estimating (so I can actually make a realistic estimation).

**Comment by nelsonic** [comment]
@RobStallion I discovered a few weeks back that I needed to improve my editing speed. 🐌 
Hence: https://github.com/nelsonic/learn-vim 🚀

**Comment by RobStallion** [comment]
I WILL be following that repo closely. I don't feel like I am at a point in my coding (yet) where I am being "held back" by my editing speed but it never hurts to be faster 👍

Plus I just like vim 👍 👍

**Comment by nelsonic** [comment]
@RobStallion can you type (_without errors_) as _fast_ as you can _think_? 🤔 
https://youtu.be/8HPF9uPllt8

**Comment by RobStallion** [comment]
@nelsonic haha I wish. What I meant is if I am trying to solve an issue, almost always it's not my typing that is the "main hold up". It's the fact that I need to figure out what is the actual cause of the issue. 

I do think that learning to use vim has improved my speed when it comes to coding things that are straightforward for me. But what I think has even more of an impact on my overall speed for "things I know how to do" are snippets. I have been trying to get into the habit of creating snippets for things I do over and over. For example, here are some of the snippets I have made for elixir in atom...
```
'.source.elixir':
  'IO.puts':
    'prefix': 'ip'
    'body': 'IO.puts(~s($1))'
  'Pipe IO.inspect':
    'prefix': 'iii'
    'body': '|> IO.inspect(label: "-----> $1")'
  'Function with args':
    'prefix': 'df'
    'body': 'def ${1:function_name}($2) do\n\t$0\nend'
  'Function with guard clause':
    'prefix': 'dfg'
    'body': 'def ${1:function_name}($2) when $3 do\n\t$0\nend'
  'Heredoc':
    'prefix': 'heredoc'
    'body': '"""\n$1\n"""'
  'Inline function with args':
    'prefix': 'idf'
    'body': 'def ${1:function_name}($2), do: $3'
  'Inline function with guard clause':
    'prefix': 'idfg'
    'body': 'def ${1:function_name}($2) when $3, do: $4'
```
I feel like no matter how fast I am at typing, I am never ever going to be as fast as the computer. Definitely doesn't mean I shouldn't get faster but I just want to also make to most out of what the computer is good at.

Vim for sure has it's place though. I find vim especially helpful for editing files and pretty much anything else that doesn't require a lot of "problem solving". It was also **SUPER** handy when I was working with the ukpostcodes.csv file.

**Comment by RobStallion** [comment]
@nelsonic 
> https://youtu.be/8HPF9uPllt8

That video was amazing. Not only is he blind but he is only 10. That is so good.

**Comment by RobStallion** [comment]
![image](https://user-images.githubusercontent.com/15571853/52118735-07e00e80-260f-11e9-9c46-fcccdd75946f.png)

This is the worst

**Comment by nelsonic** [comment]
Yeah, couldn't agree more. [**`#Nagware`**](https://en.wikipedia.org/wiki/Shareware#Nagware)

> **Note**: I'm guessing that from their cohort/abandonment analysis they have determined that if they _don't_ "nag" people to "Upgrade", nobody does or people "drop off" because the "_core_" product isn't good enough to make people ["_effective_"](https://github.com/nelsonic/nelsonic.github.io/issues/651)...? 

Personally I cannot _stand_ apps that "nag" me to do _anything_ even if it's for my own _benefit_!
Modals are _evil horrible_ UX. Constantly _forcing_ me to close something is _infuriating_. 
There are _so many_ better user-messaging UI techniques ...!

**Comment by RobStallion** [comment]
I think I have decided that I don't like toggl. I have barely used it the last few days (can see exactly which days in the app) because it just doesn't make me want to come back to it. 

I decided to download the desktop app to see if it would be a better experience. First things first, I think it is less clear than the site (which I'm shocked at tbh because I don't think that the site is very clear to begin with (I'm not talking about the metric pages, just the timer page)). 

One big plus of the desktop app that I noticed though is that it can account for time that you were afk. So a little while ago I stopped working to help with a fac question but forgot to turn off my timer. When I came back to my computer, toggl desktop had a little pop up asking if I wanted to "remove" the period I was away for (I think it could tell when my laptop was locked). 

This is a cool feature that will not be able to be reproduced in the browser (as far as I am aware anyway). That being said, it wouldn't be a feature that would be a winner for me anyway (I mean it wouldn't make or break me using the app).

**Comment by nelsonic** [comment]
Channel that _frustration_ with Toggl into building a _better_ way of tracking your effort. 😉

**Comment by RobStallion** [comment]
My favourite toggl feature by far... (enough to take me away from the desktop app back to the browser)

### active
![image](https://user-images.githubusercontent.com/15571853/52713125-63ef4f00-2f8e-11e9-840d-63ab3ca7006b.png)

### not
![image](https://user-images.githubusercontent.com/15571853/52713243-b29ce900-2f8e-11e9-8e9b-4176c32b5a44.png)

- Reminds me to actually use it because I see it(good step to have while I am trying to make this a habit)
- I can see at a glance how long I have been working on a task without even switching tabs. 
- I can see if the timer is active or not (above images)

**Comment by nelsonic** [comment]
@robstallion worth adding this insight to the original issue. 😉

**Comment by nelsonic** [comment]
What was your conclusion on this?
Did you move it to a more appropriate place (than “chess”)?
(I only ask out of curiosity/learning...)

**Comment by RobStallion** [comment]
@nelsonic I don't think that I really came to much of a conclusion with this issue. I think I just closed it as Toggl had been frustrating me (which is a really really bad reason to close an issue. Especially when I think that the idea of tracking my time could be helpful going forward).

I am reopening this issue and will start tracking my time again so that I can actually try and come to some conclusions about this.

**Comment by nelsonic** [comment]
@RobStallion looking forward to reading your insights. 👍 
I 100% agree that the Toggl UI/UX is _frustrating_ (AF!) and we could do a _much_ better job of it! 😉

**Comment by RobStallion** [comment]
I want a way of seeing if I have been working on / stuck on one part of a particular issue for too long.

Right now I think that this might be a "me" problem.
For example, I am working on an issue that I have estimated will take 4 hours but some time into that issue I hit a bug that I didn't expect. I continue working on the issue trying to solve the bug and finish the issue. This eats into my time massively and before I know it I am over the estimated time. Not only this, but I do not know how long I have actually been on the bug for (specifically the bug, not the issue as a whole). 

The reason I would like to know how long I have been on a bug for is so I can make a more informed decision on if I should move onto something else or not. 

Capturing my thoughts for now. Will come back to this later in more detail

**Comment by nelsonic** [comment]
@RobStallion being "stuck" on something is not a "you" problem at _all_.
What you are describing is two issues: 
1. A task that is estimated at **`T4h`** is not sufficiently broken down and will almost _always_ take longer.
  Unless the person is _incredibly_ focussed they will (_generally_) not be _able_ to work for **4h** straight. 
  This is not specific to _you_ (Rob) but is applicable to _everyone_. Cal Newport manages around **5h** of "Deep Work" per day but he breaks it down as much as he can to stay in the ["flow state"](https://github.com/nelsonic/nelsonic.github.io/issues/689) for as long as possible. 
  (_I am guilty of doing estimating tasks as `T1d` ... however I feel that applying this label should **automatically** require the person to create sub-tasks to break down the work further ..._)
2. Whenever you hit a bug or unforeseen hurdle/requirement it should _always_ be noted down as a _separate_ item on the Task list. 

Look forward to reading your further insights on this. 👍
